Remix v2

frandiox opened the pull request:
FYI We'll release v2 support in our next major bump to avoid breaking changes for users in minor/patch releases.

---

This is just to track the required changes for supporting Remix v2.

~I had to add a few `@ts-ignore` due to a bug in the types: https://github.com/remix-run/remix/issues/7246~ Fixed!
Should be fixed for v2 stable I think.

Todos:
- [x] Update demo-store, hello-world and examples
- [x] Remove old code for v1
- [x] Remove bug related `@ts-ignore` comments
- [x] Update support for JSDoc when #1334 is merged


~Note: the preview environment in this PR shows the demo-store and it has unrelated errors due to the new CSP~

frandiox committed:
Update Remix versions to 2.0.0-pre.3

frandiox committed:
Fix types in Hydrogen

frandiox committed:
Fix type exports in Oxygen adapter

frandiox committed:
Mock v2 flags in CLI

frandiox committed:
Rename V2_MetaFunction type in skeleton

frandiox committed:
Remove CSS nesting to avoid ESBuild warnings

frandiox committed:
Remove old error and catch boundaries

frandiox committed:
Add small wrapper for root data to fix type unknown

frandiox committed:
ts-ignore jsonify errors
https://github.com/remix-run/remix/issues/7246

frandiox committed:
Remove vs flags from template

frandiox committed:
Update to Remix 2.0.0

frandiox committed:
Update skeleton types to v2

frandiox committed:
Update hello-world types to v2

frandiox committed:
Update demo-store types to v2

frandiox committed:
Update examples types to v2

frandiox committed:
Remove future v2 flags

frandiox committed:
Minor type update

frandiox committed:
Merge branch 'main' into fd-remix-2

frandiox committed:
Remove v2 flags from CLI

frandiox committed:
Remix deprecated tsconfig in config

frandiox committed:
Fix typecheck

frandiox committed:
Stop injecting booleans in remix config. Fix tests

frandiox committed:
Package lock

frandiox committed:
Fix types

frandiox committed:
Remove unused tests

frandiox committed:
Cleanup

frandiox committed:
Merge branch 'main' into fd-remix-2

frandiox committed:
Update package-lock

frandiox committed:
Update Remix versions to latest nightly

frandiox committed:
Fix optimistic-ui for Remix v2

frandiox committed:
Update package-lock

frandiox committed:
Remove ts-ignore comments related to JsonifyObject error

frandiox committed:
Add missing reference to eslint package in skeleton

frandiox committed:
Update to Remix 2.1.0

frandiox committed:
Merge branch 'main' into fd-remix-2

frandiox committed:
Fix SerializeFrom in JSDoc

frandiox committed:
Fix typedefs in JSDoc

frandiox committed:
Use SerializeFrom in demo-store

frandiox committed:
Changesets

frandiox commented on templates/demo-store/app/routes/($locale).products.$productHandle.tsx:
```diff
@@ -59,7 +59,7 @@ export async function loader({params, request, context}: LoaderArgs) {
 
   if (!product.selectedVariant) {
-    return redirectToFirstVariant({product, request});
+    throw redirectToFirstVariant({product, request});
```
This makes returned types work better, and I think it doesn't change the behavior, right?

frandiox commented on templates/skeleton/app/root.tsx:
```diff
@@ -58,7 +60,12 @@ export function links() {
 }
 
-export async function loader({context}: LoaderArgs) {
+export const useRootLoaderData = () => {
+  const [root] = useMatches();
+  return root?.data as SerializeFrom<typeof loader>;
+};
```
I'm adding this small abstraction to enhance types. Thoughts?

frandiox committed:
Support generics in JSDoc

frandiox committed:
Set Remix as a non-pinned peer dependency

frandiox committed:
Changesets

blittle commented on .changeset/unlucky-stingrays-live.md:
````diff
@@ -0,0 +1,23 @@
+}
+```
+
+This way, you can import `useRootLoaderData()` anywhere in your app and get the correct type for the data returned by the root loader.
````
This is cool, I didn't know about it.

frandiox committed:
Update .changeset/smart-ways-destroy.md

Co-authored-by: Bret Little <bret.little@shopify.com>

frandiox committed:
Merge branch 'main' into fd-remix-2

wizardlyhel reviewed (approved):
I'm good with `useRootLoaderData`

I think we should note breaking change in changeset?

* `V2_MetaFunction` rename to `MetaFunction`
* `LoaderArgs` rename to `LoaderFunctionArgs`

frandiox committed:
Merge branch 'main' into fd-remix-2

frandiox committed:
Remove react dep again

frandiox committed:
Remove template guidelines for Remix v1

frandiox committed:
Minor fix

frandiox committed:
Update changeset

frandiox commented:
>I think we should note breaking change in changeset?

I had links to the Remix release notes in the demo-store changeset but I've now added the same links to the other packages. Also added a couple of common examples 👍
